service/controller/app/v1/resource/chart/current.go

```diff
@@ -0,0 +1,35 @@
+package chart
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	name := key.AppName(cr)
+	g8sClient, err := r.kubeConfig.NewG8sClientForApp(ctx, cr)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	chart, err := g8sClient.ApplicationV1alpha1().Charts(r.watchNamespace).Get(name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find chart %#q", name))
```

Comment:
The logging here is cool. Can we also have some for before and after the action? The visibility on what is going on is pretty useful, especially when developing new resources. Check our `fmt` docs or other resource implementations.

Reply:
it make sense, I will create other msgs before/after action as well.

Reply:
@xh3b4sd I need your approval, PTAL :)